torchmetrics/detection/map.py

```diff
@@ -279,20 +291,339 @@ def update(self, preds: List[Dict[str, Tensor]], target: List[Dict[str, Tensor]]
+        """
+        gt = self.groundtruth_boxes[id]
+        dt = self.detection_boxes[id]
+        gt_lbl_mask = self.groundtruth_labels[id] == class_id
+        dt_lbl_mask = self.detection_labels[id] == class_id
+        if len(dt_lbl_mask) == 0 or len(dt_lbl_mask) == 0:
+            return torch.tensor([])
+        gt = gt[gt_lbl_mask]
+        dt = dt[dt_lbl_mask]
+        if len(gt) == 0 or len(dt) == 0:
+            return torch.tensor([])
+
+        # Sort by scores and use only max detections
+        scores = self.detection_scores[id]
+        scores_filtered = scores[self.detection_labels[id] == class_id]
+        inds = torch.argsort(scores_filtered, descending=True)
+        dt = dt[inds]
+        if len(dt) > max_det:
+            dt = dt[:max_det]
+
+        # generalized_box_iou
+        ious = box_iou(dt, gt)
+        return ious
+
+    def _evaluate_image(self, id: int, class_id: int, area_range: List[int], max_det: int, ious: Tensor) -> Dict:
+        """Perform evaluation for single class and image.
+
+        Args:
+            id:
+                Image Id, equivalent to the index of supplied samples.
+            class_id:
+                Class Id of the supplied groundtruth and detection labels.
+            area_range:
+                List of lower and upper bounding box area threshold.
+            max_det:
+                Maximum number of evaluated detection bounding boxes.
+            ious:
+                IoU reults for image and class.
+        """
+        gt = self.groundtruth_boxes[id]
+        dt = self.detection_boxes[id]
+        gt_lbl_mask = self.groundtruth_labels[id] == class_id
+        dt_lbl_mask = self.detection_labels[id] == class_id
+        if len(dt_lbl_mask) == 0 or len(dt_lbl_mask) == 0:
+            return None
+        gt = gt[gt_lbl_mask]
+        dt = dt[dt_lbl_mask]
+        if len(gt) == 0 and len(dt) == 0:
+            return None
+
+        areas = box_area(gt)
+        ignore_area = (areas < area_range[0]) | (areas > area_range[1])
+
+        # sort dt highest score first, sort gt ignore last
+        ignore_area_sorted, gtind = torch.sort(ignore_area)
+        gt = gt[gtind]
+        scores = self.detection_scores[id]
+        scores_filtered = scores[dt_lbl_mask]
+        scores_sorted, dtind = torch.sort(scores_filtered, descending=True)
+        dt = dt[dtind]
+        if len(dt) > max_det:
+            dt = dt[:max_det]
+        # load computed ious
+        ious = ious[id, class_id][:, gtind] if len(ious[id, class_id]) > 0 else ious[id, class_id]
+
+        T = len(self.iou_thresholds)
+        G = len(gt)
+        D = len(dt)
+        gt_matches = torch.zeros((T, G), dtype=torch.bool)
+        dt_matches = torch.zeros((T, D), dtype=torch.bool)
+        gt_ignore = ignore_area_sorted
+        dt_ignore = torch.zeros((T, D), dtype=torch.bool)
+        if len(ious) > 0:
+            for tind, t in enumerate(self.iou_thresholds):
+                for d in range(D):
```

Comment:
for readability pls name variables with at least three letters... otherwise, it is impossible to use any replace tool in future

Reply:
refactor method exists ;)
But I get the point